pilot/platform/kube/inject/file.go

```diff
@@ -0,0 +1,170 @@
+// Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package inject
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"reflect"
+
+	"github.com/ghodss/yaml"
+
+	meshconfig "istio.io/api/mesh/v1alpha1"
+	// TODO(nmittler): Remove this
+	_ "github.com/golang/glog"
+
+	appsv1beta1 "k8s.io/api/apps/v1beta1"
+	batchv1 "k8s.io/api/batch/v1"
+	"k8s.io/api/batch/v2alpha1"
+	"k8s.io/api/core/v1"
+	"k8s.io/api/extensions/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	yamlDecoder "k8s.io/apimachinery/pkg/util/yaml"
+)
+
+var (
+	kinds = []struct {
+		groupVersion schema.GroupVersion
+		obj          runtime.Object
+		resource     string
+		apiPath      string
+	}{
+		{v1.SchemeGroupVersion, &v1.ReplicationController{}, "replicationcontrollers", "/api"},
+
+		{v1beta1.SchemeGroupVersion, &v1beta1.Deployment{}, "deployments", "/apis"},
+		{v1beta1.SchemeGroupVersion, &v1beta1.DaemonSet{}, "daemonsets", "/apis"},
+		{v1beta1.SchemeGroupVersion, &v1beta1.ReplicaSet{}, "replicasets", "/apis"},
+
+		{batchv1.SchemeGroupVersion, &batchv1.Job{}, "jobs", "/apis"},
+		{v2alpha1.SchemeGroupVersion, &v2alpha1.CronJob{}, "cronjobs", "/apis"},
+		// TODO JobTemplate requires different reflection logic to populate the PodTemplateSpec
```

Comment:
Nit: If you know the details for this TODO, can you flesh it out with links or other context? There is some logic for CronJobs below, but I'm not sure what more we may be missing.

Reply:
The TODO is out of date. This was fixed in https://github.com/istio/istio/pull/1619. I'll remove it.